### PR TITLE
Fix SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder"

### PR DESCRIPTION
**Pull request by claudio4j**
camel-k-maven-logging from camel-k-runtime will bring slf4j-api 2.0.
slf4j-api 1.7 originally from maven zip should be removed

Fix https://github.com/apache/camel-k/issues/4882

The [camel-k-runtime fix 1100](https://github.com/apache/camel-k-runtime/pull/1100) should be merged first

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by squakez — COMMENTED** [review]
LGTM. I'm converting to a draft as we want to retest and merge this once the new default runtime version with the fix is available.

However this shows a runtime dependency we need to remove. This was something we included in the runtime when we had a single runtime. It makes sense to move any required dependency off the runtime now that we want to make it independent. Going to open a follow up issue. Or do you think it is something you want to tackle now? we should be able to easily move back the logging module into Camel K and let any operator configuration to be inside the right repository.

**Comment by claudio4j** [comment]
the runtime dependency is not good, we can deal with this exclusively on camel-k operator side, by having the maven_overlay script to check the slf4j / logback compatibility and add the missing slf4j-2.0.jar, wdyt ?

**Comment by squakez** [comment]
Yes. This is kind of how it was managed before [1]. However, we used to have a static set of dependencies. We should have instead an internal maven project in order to have dependabot and all the stuff on it. 

[1] https://github.com/apache/camel-k/pull/3376

**Comment by claudio4j** [comment]
A proposal would be to move the camel-k-runtime/camel-k-maven-logging to camel-k/java project.

**Review by squakez — CHANGES_REQUESTED** [review]
I think the crds and the logging have to be separate maven projects. During release process we publish the CRDs, while we're not willing to publish the maven logging dependency (which is only used internally). For the rest, it looks fine.

**Comment by claudio4j** [comment]
Do you mean to remove the parent java/pom.xml ?

**Comment by squakez** [comment]
> Do you mean to remove the parent java/pom.xml ?

Yes. CRDs and Logging have to be separate poms.

File: script/Makefile [hunk 1]
```diff
@@ -430,7 +430,7 @@ check-licenses:
 maven-overlay:
 	@echo "####### Preparing maven dependencies bundle..."
 	mkdir -p build/_maven_overlay
-	./script/maven_overlay.sh -s "$(STAGING_RUNTIME_REPO)" -d "$(CAMEL_K_RUNTIME_DIR)" $(DEFAULT_RUNTIME_VERSION) build/_maven_overlay
+	./script/maven_overlay.sh build/_maven_overlay
```
Comment (squakez) [inline script/Makefile:433]: Maybe it misses a `mvn clean package` before calling the script? or alternatively it should be executed in the script.

**Review by squakez — COMMENTED** [review]
I think you're deleting the symbolic links to the crds and the license/notice files. They must be part of the CRD project.

File: script/Makefile (hunk 1 above)
Comment (claudio4j) [reply script/Makefile:433]: The maven_overlay.sh contains the clean operation.

Comment (claudio4j) [reply script/Makefile:433]: done

**Comment by claudio4j** [comment]
Somehow the symlink got lost, I will link them.

**Comment by claudio4j** [comment]
I think this fix is important for 2.1.x, I will provide a PR soon.

**Review by squakez — APPROVED** [review]

**Pull request merged**